### PR TITLE
Add tests for tools

### PR DESCRIPTION
**Pull request by 0xArdi**
This PR adds tests for tools.
To run the tests, you can use:
```
tox -e check-tools
```
or alternatively:
```
pytest tests/
```

### CODE REVIEW

File: packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py [hunk 1]
```diff
@@ -315,21 +450,24 @@ def get_urls_from_queries(
...
+            for url in search_google(
+                query=query,
+                api_key=api_key,
+                engine=engine,
+                num=num,
+            ):
+                results.append(url)
+        except Exception:
+            pass
     unique_results = list(set(results))
```
Comment (cyberosa) [inline packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:463]: what if all queries fail and you have no urls?

File: packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py [hunk 2]
```diff
@@ -341,79 +479,86 @@ def extract_text_from_pdf(url: str, num_words: Optional[int] = None) -> str:
...
+    else:
+        text = " ".join(text.split())
+
+    doc = Document(text=text, url="")
+    return doc
+
+
+def extract_texts(urls: List[str], num_words: Optional[int] = None) -> List[Document]:
+    """Extract texts from URLs with improved error handling, excluding failed URLs."""
     extracted_texts = []
```
Comment (cyberosa) [inline packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:515]: if the list of urls is empty then there is nothing to batch, you could return an empty list and avoid the loop

File: tests/test_tools.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    tools = prediction_request_reasoning.ALLOWED_TOOLS
+    models = prediction_request_reasoning.ALLOWED_MODELS
+    prompts = [
+        'Please take over the role of a Data Scientist to evaluate the given question. With the given question "Will Apple release iPhone 17 by March 2025?" and the `yes` option represented by `Yes` and the `no` option represented by `No`, what are the respective probabilities of `p_yes` and `p_no` occurring?'
+    ]
+    tool_module = prediction_request_reasoning
+
+
+class TestPredictionCOT(BaseToolTest):
+    """Test Prediction RAG."""
```
Comment (cyberosa) [inline tests/test_tools.py:137]: I guess you mean Test Prediction COT

**Review by cyberosa — APPROVED** [review]

File: packages/napthaai/customs/prediction_request_rag_claude/component.yaml
Comment (0xArdi) [inline packages/napthaai/customs/prediction_request_rag_claude/component.yaml:1]: why deleted?

File: packages/napthaai/customs/prediction_request_rag_claude/prediction_request_rag_claude.py
Comment (0xArdi) [inline packages/napthaai/customs/prediction_request_rag_claude/prediction_request_rag_claude.py:1]: same

File: packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py (hunk 1 above)
Comment (richardblythman) [reply packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:463]: Not they ever all fail

Comment (richardblythman) [reply packages/napthaai/customs/prediction_request_reasoning/prediction_request_reasoning.py:463]: I haven't seen it happen at least

File: packages/napthaai/customs/prediction_request_rag_claude/component.yaml
Comment (richardblythman) [reply packages/napthaai/customs/prediction_request_rag_claude/component.yaml:1]: The whole tool has been deleted. Claude can now be used in prediction_request_rag.

**Pull request merged**